src/pages/RegisterStudents.vue

```diff
@@ -653,15 +667,35 @@ const getOrgId = async (orgType, orgName, selectedDistrict = null, selectedSchoo
 
   // If not in cache, fetch and cache the result
   try {
-    const org = await fetchOrgByName(orgType, orgName, { value: selectedDistrict }, { value: selectedSchool });
-    if (org[0]?.id) {
-      orgCache.value[orgType].set(cacheKey, org[0].id);
-      return org[0].id;
-    }
+    // Fetch user's available orgs of type orgType
+    const userAdminOrgs = await orgFetchAll(
+      orgType,
+      selectedDistrict,
+      selectedSchool,
+      orderByDefault,
+      isSuperAdmin,
+      adminOrgs,
+      ['id', 'name', 'districtId', 'schoolId', 'schools', 'classes'],
+    );
+
+    // Cache orgs in case we need them for a subsequent call
+    userAdminOrgs.forEach((org) => {
+      const cacheKey =
+        orgType === 'schools'
+          ? `${org.name}-${selectedDistrict}`
+          : orgType === 'classes'
+          ? `${org.name}-${selectedSchool}`
+          : org.name;
+      orgCache.value[orgType].set(cacheKey, org.id);
+    });
+
+    // Find org with name orgName
+    const org = userAdminOrgs.find((o) => o.name === orgName);
```

Comment:
Just a final thought: this could, theoretically fail, if the uploaded CSV contains a stray empty space or has inconsistent casing. As mentioned during our call, most of this work should be offloaded to the backend, but for now, a simple improvement could already help: 

```suggestion
    const org = userAdminOrgs.find(
      (o) => o.name.trim().toLowerCase() === orgName.trim().toLowerCase()
    );
```